### PR TITLE
feat: Add new example demonstrating how to configure prefix delegation to increase the amount of available IP address

### PR DESCRIPTION
**Pull request by bryantbiggs**
### What does this PR do?
- Add new example demonstrating how to configure prefix delegation to increase the amount of available IP address
- Update EKS managed node groups AL2 userdata to render templates in terraform and remove logic not used in rendered userdata from what is sent to the launch template. This keeps the userdata cleaner (there are size restrictions) and only to what is in use by the current settings 
 
### Motivation
- Demonstrate how users can enable https://docs.aws.amazon.com/eks/latest/userguide/cni-increase-ip-addresses.html to better utilize the available IP space

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [ ] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by vara-bonthu — APPROVED** [review]
This looks great! Very much needed example for customizing VPC CNI Daemon Set and the MEX PODS for the Managed Node groups 👍🏼

File: examples/ipv4-prefix-delegation/variables.tf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+variable "deploy_sample_app" {
+  default     = false
+  type        = bool
+  description = "Deploy a sample app"
+}
```
Comment (vara-bonthu) [inline examples/ipv4-prefix-delegation/variables.tf:5]: This variable is not used anywhere

Comment (bryantbiggs) [reply examples/ipv4-prefix-delegation/variables.tf]: thanks for catching that - looks like I had some leftover copy+pasta to clean up!

File: examples/ipv4-prefix-delegation/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
+# EKS Cluster w/ Prefix Delegation
+
+This example shows how to provision an EKS cluster with prefix delegation enabled for increasing the number of available IP addresses for the EC2 nodes utilized.
+
+- [Docuemtation](https://docs.aws.amazon.com/eks/latest/userguide/cni-increase-ip-addresses.html)
```
Comment (FernandoMiguel) [inline examples/ipv4-prefix-delegation/README.md:5]: Typo `Docuemtation`

**Pull request merged**